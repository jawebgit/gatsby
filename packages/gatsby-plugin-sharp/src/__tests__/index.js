const path = require(`path`)

const {
  queueImageResizing,
  base64,
  responsiveSizes,
  responsiveResolution,
} = require(`../`)

describe(`gatsby-plugin-sharp`, () => {
  const args = {
    duotone: false,
    grayscale: false,
    rotate: false,
  }
  const absolutePath = path.join(__dirname, `images/test.png`)
  const file = {
    id: `${absolutePath} absPath of file`,
    absolutePath,
    extension: `png`,
    internal: {
      contentDigest: `1234`,
    },
  }

  describe(`responsiveSizes`, () => {
    it(`includes responsive image properties, e.g. sizes, srcset, etc.`, async () => {
      const result = await responsiveSizes({ file })

      expect(result).toMatchSnapshot()
    })

    it(`adds pathPrefix if defined`, async () => {
      const pathPrefix = `/blog`
      const result = await responsiveSizes({
        file,
        args: {
          pathPrefix,
        },
      })

      expect(result.src.indexOf(pathPrefix)).toBe(0)
      expect(result.srcSet.indexOf(pathPrefix)).toBe(0)
    })
  })

  describe(`base64`, () => {
    it(`converts image to base64`, async () => {
      const result = await base64({
        file,
        args,
      })

      expect(result).toMatchSnapshot()
    })
  })
})
