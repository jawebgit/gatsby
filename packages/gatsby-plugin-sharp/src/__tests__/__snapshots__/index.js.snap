// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gatsby-plugin-sharp base64 converts image to base64 1`] = `
Object {
  "aspectRatio": 1,
  "height": 20,
  "originalName": undefined,
  "src": "data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABQAAAAUCAIAAAAC64paAAAACXBIWXMAAAsTAAALEwEAmpwYAAAAGElEQVQ4y2P4TwFgGNU8qnlU86jmgdUMAOBEq42KgAyMAAAAAElFTkSuQmCC",
  "width": 20,
}
`;

exports[`gatsby-plugin-sharp responsiveSizes includes responsive image properties, e.g. sizes, srcset, etc. 1`] = `
Object {
  "aspectRatio": 1,
  "base64": "data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABQAAAAUCAIAAAAC64paAAAACXBIWXMAAAsTAAALEwEAmpwYAAAAGElEQVQ4y2P4TwFgGNU8qnlU86jmgdUMAOBEq42KgAyMAAAAAElFTkSuQmCC",
  "originalImg": "/static/1234-7fd61.png",
  "originalName": undefined,
  "sizes": "(max-width: 800px) 100vw, 800px",
  "src": "/static/1234-7fd61.png",
  "srcSet": "/static/1234-7fd61.png 1w",
}
`;
